### PR TITLE
feat(ui): Add source code navbar and dir panel

### PR DESCRIPTION
**Pull request by liangfung**

### CODE REVIEW

**Comment by wsxiaoys** [comment]
Could you remove the `name` header?

image

Otherwise LGTM!

**Comment by wsxiaoys** [comment]
Please also fix the lint: https://github.com/TabbyML/tabby/actions/runs/7655342170/job/20861184075?pr=1295

File: crates/llama-cpp-bindings/llama.cpp
Comment (wsxiaoys) [inline crates/llama-cpp-bindings/llama.cpp:1]: please revert change to this file

**Comment by liangfung** [comment]
> Could you remove the `name` header?

done

**Comment by liangfung** [comment]
> Could you remove the `name` header?

`name` removed
image

**Pull request merged**